### PR TITLE
Xdmod9.0/appkernel data aggregation for hichart

### PR DESCRIPTION
**Pull request by nsimakov**
Added data aggregation for plotting to overcome hichart limitations on large datasets.

Was:
ak_usage_Before
Now:
ak_usage_After
Was:
data_explorer_Before
Now:
data_explorer_After

There are also some conversions to handle AKRR 2.0

### CODE REVIEW

**Comment by ryanrath** [comment]
Just a quick question, do you have an editor that you prefer using? A number of them support setting up a set of code format guidelines that you can then have automatically applied to whatever bit of code that you're working on. I think that would make the vast majority of the comments ( minus the parameterized query ones ) just a single key stroke per file to update.

File: classes/AppKernel/AppKernelDb.php [hunk 1]
```diff
@@ -10,20 +11,21 @@
 
 // Include data class definitions
 
-require_once("InstanceData.php");
+require_once "InstanceData.php";
```
[the remark this answers is not in the document]
Comment (nsimakov) [reply classes/AppKernel/AppKernelDb.php:14]: I don't know let's check will it work

File: classes/AppKernel/AppKernelDb.php [hunk 2]
```diff
@@ -133,7 +132,7 @@ public function storeIngestionLogEntry(IngestionLogEntry $log)
             "INSERT INTO ingester_log
             (source, url, num, last_update, start_time, end_time, success, message, reportobj)
             VALUES (?, ?, ?, FROM_UNIXTIME(?), FROM_UNIXTIME(?), FROM_UNIXTIME(?), ?, ?, COMPRESS(?))";
-            $params = array($log->source,
+        $params = array($log->source,
```
[the remark this answers is not in the document]
Comment (nsimakov) [reply classes/AppKernel/AppKernelDb.php:135]: done

File: classes/AppKernel/AppKernelDb.php [hunk 3]
```diff
@@ -142,11 +141,13 @@ public function storeIngestionLogEntry(IngestionLogEntry $log)
             $log->success,
             $log->message,
             $log->reportObj);
-        $numRows = $this->db->execute($sql, $params);
-        if ( 1 == $numRows ) $this->log("Added entry to ingestion log");
+            $numRows = $this->db->execute($sql, $params);
```
[the remark this answers is not in the document]
Comment (nsimakov) [reply classes/AppKernel/AppKernelDb.php:144]: ok

File: classes/AppKernel/AppKernelDb.php [hunk 4]
```diff
@@ -142,11 +141,13 @@ public function storeIngestionLogEntry(IngestionLogEntry $log)
...
+        if (1 == $numRows ) {
```
[the remark this answers is not in the document]
Comment (nsimakov) [reply classes/AppKernel/AppKernelDb.php:145]: ok

File: classes/AppKernel/AppKernelDb.php [hunk 5]
```diff
@@ -806,105 +820,110 @@ public function loadResources(array $criteria = NULL)
...
-                $groupBy = 'num_units';
-                $orderBy = 'num_units';
-                break;
-            default:
-                break;
-            }  // switch ($criteria['group_by'])
-        }  // if ( array_key_exists('group_by', $criteria) ... )
-
-        if ( 0 != count($sqlCriteria) )
+        if (0 != count($sqlCriteria) ) {
```
[the remark this answers is not in the document]
Comment (nsimakov) [reply classes/AppKernel/AppKernelDb.php:906]: done

File: classes/AppKernel/AppKernelDb.php [hunk 6]
```diff
@@ -806,105 +820,110 @@ public function loadResources(array $criteria = NULL)
...
         $sql = "SELECT * ".($debugMode ?'':", unix_timestamp(min(start_time)) as start_ts, unix_timestamp(max(end_time)) as end_ts ")."  FROM " . ( $debugMode ? "v_tree_debug" : "a_tree" );
 
-        if ( NULL !== $criteria && 0 != count($criteria) )
-        {
-            foreach ( $criteria as $key => $value )
-            {
-                if ( NULL === $value ) continue;
-                switch ($key)
-                {
+        if (null !== $criteria && 0 != count($criteria) ) {
```
[the remark this answers is not in the document]
Comment (nsimakov) [reply classes/AppKernel/AppKernelDb.php:833]: ok

File: classes/AppKernel/AppKernelDb.php (hunk 5 above)
[the remark this answers is not in the document]
Comment (nsimakov) [reply classes/AppKernel/AppKernelDb.php:906]: ok

File: classes/AppKernel/AppKernelDb.php [hunk 7]
```diff
@@ -926,109 +945,114 @@ public function loadTreeLevel(array $criteria = NULL)
...
         $sql = "SELECT * FROM " . ( $debugMode ? "v_tree_debug" : "a_tree" );
 
-        if ( NULL !== $criteria && 0 != count($criteria) )
-        {
-            foreach ( $criteria as $key => $value )
-            {
-                if ( NULL === $value ) continue;
-                switch ($key)
-                {
+        if (null !== $criteria && 0 != count($criteria) ) {
```
[the remark this answers is not in the document]
Comment (nsimakov) [reply classes/AppKernel/AppKernelDb.php:958]: ok

File: classes/AppKernel/AppKernelDb.php [hunk 8]
```diff
@@ -926,109 +945,114 @@ public function loadTreeLevel(array $criteria = NULL)
...
-                $groupBy = 'num_units';
-                $orderBy = 'num_units';
-                break;
-            default:
-                break;
-            }  // switch ($criteria['group_by'])
-        }  // if ( array_key_exists('group_by', $criteria) ... )
-
-        if ( 0 != count($sqlCriteria) )
+        if (0 != count($sqlCriteria) ) {
```
[the remark this answers is not in the document]
Comment (nsimakov) [reply classes/AppKernel/AppKernelDb.php:1031]: done

File: classes/AppKernel/AppKernelDb.php [hunk 9]
```diff
@@ -1057,73 +1081,71 @@ public function getDataPointsQuery(array $criteria = NULL)
...
-                    $sqlParams[] = $value;
-                    break;
-                case 'end':
-                    if ( $metadataOnly ) {
-                        $sqlCriteria[] = "(UNIX_TIMESTAMP(?)+86399) >= UNIX_TIMESTAMP(start_time)";
-                    } else {
-                        if($datesAsEpoch)
-                        {
-                            $sqlCriteria[] = "? >= collected";
+        if (null !== $criteria && 0 != count($criteria) ) {
```
[the remark this answers is not in the document]
Comment (nsimakov) [reply classes/AppKernel/AppKernelDb.php:1084]: done

File: classes/AppKernel/AppKernelDb.php [hunk 10]
```diff
@@ -1221,11 +1244,14 @@ public function loadAppKernelInstances($appKernelDefId = NULL, $resourceId = NUL
...
+        if (null !== $appKernelDefId ) {
+            $criteriaList[] = "i.ak_def_id = $appKernelDefId";
+        }
+        if (null !== $resourceId ) {
+            $criteriaList[] = "resource_id = $resourceId";
+        }
 
-        if ( 0 != count($criteriaList) )
-        {
+        if (0 != count($criteriaList) ) {
```
[the remark this answers is not in the document]
Comment (nsimakov) [reply classes/AppKernel/AppKernelDb.php:1254]: ok

File: classes/AppKernel/AppKernelDb.php [hunk 11]
```diff
@@ -1691,74 +1726,61 @@ private function addDebugData(InstanceData $ak, $replace, $dryRunMode=false)
...
-        {
-            
         }
-        if($add_to_a_data)
-        {
+        if($add_to_a_data) {
             //a_data and a_tree
-            if($ak->db_resource_visible && $ak->db_ak_def_visible && $ak->status=='success')
-            {
+            if($ak->db_resource_visible && $ak->db_ak_def_visible && $ak->status=='success') {
```
[the remark this answers is not in the document]
Comment (nsimakov) [reply classes/AppKernel/AppKernelDb.php:1755]: ok

File: html/.eslintrc.json [hunk 12]
```diff
@@ -0,0 +1,19 @@
+{
```
[the remark this answers is not in the document]
Comment (nsimakov) [reply html/.eslintrc.json:1]: it should be there or eslint should complain on unknown namespaces and travis will fail styling checking

**Comment by nsimakov** [comment]
Surely I use good editor, but I was assuming php code sniffer with settings from xdmod-qa provides enough coverage for our coding standard and use it for linting. On the other hand this is largely issue solving PR with a fair amount of styling fixes.

File: classes/AppKernel/AppKernelDb.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (nsimakov) [reply classes/AppKernel/AppKernelDb.php:14]: Due to multiple classes in InstanceData autoloading is not working

File: classes/AppKernel/AppKernelDb.php [hunk 13]
```diff
@@ -1513,15 +1536,15 @@ public function activeAppKernels(array $appKernels)
             {
                 foreach ( $resourceAkList as $akName )
                 {
-                    if ( 0 !== strpos($akName, $basename) ) continue;
+                    if (0 !== strpos($akName, $basename) ) {
```
[the remark this answers is not in the document]
Comment (nsimakov) [reply classes/AppKernel/AppKernelDb.php:1541]: the former, $akName is <$basename>.<number of nodes>

**Pull request merged**